src/components/media/controls/controls.tsx

```diff
@@ -0,0 +1,274 @@
+import './controls.css'
+
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp'
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Icon from '@mui/material/Icon'
+import IconButton from '@mui/material/IconButton'
+import LinearProgress from '@mui/material/LinearProgress'
+import Slider from '@mui/material/Slider'
+import Typography from '@mui/material/Typography'
+import React, { useState } from 'react'
+
+import { formatDurationTime } from '../../helper'
+
+interface MediaIconButtonProps {
+  onClick: () => void
+  action: 'play' | 'pause' | 'forward' | 'replay' | 'volumeUp' | 'volumeOff'
+  className?: string
+}
+
+export function MediaIconButton(props: MediaIconButtonProps) {
+  const controls = {
+    play: { symbol: 'play_circle', label: 'play' },
+    pause: { symbol: 'pause_circle', label: 'pause' },
+    forward: { symbol: 'forward_10', label: 'forward ten seconds' },
+    replay: { symbol: 'replay_10', label: 'replay ten seconds' },
+    volumeUp: { symbol: 'volume_up', label: 'mute' },
+    volumeOff: { symbol: 'volume_off', label: 'unmute' },
+  }
+  return (
+    <IconButton
+      onClick={props.onClick}
+      aria-label={`click to ${controls[props.action].label}`}
+      className={props.className}
+      data-cy={`${props.action}-button`}
+    >
+      <Icon color="primary">
+        <span className="material-symbols-rounded">
+          {controls[props.action].symbol}
+        </span>
+      </Icon>
+    </IconButton>
+  )
+}
+
+interface MediaControls {
+  mediaElement: HTMLMediaElement
+}
+
+export function ProgressSlider(props: MediaControls) {
+  const formattedElapsedTime = formatDurationTime(
+    props.mediaElement.currentTime
+  )
+
+  function handleTimeChange(event: Event, newValue: number | number[]) {
+    props.mediaElement.currentTime = newValue as number
+  }
+
+  function renderBufferedProgressBar() {
+    const buffered = props.mediaElement.buffered
+    const percentMultiple = 100
+    let secondsBuffered = 0
+
+    // // This loop calculates the total seconds buffered up to the current playback time by iterating through the buffered time ranges in reverse order. It finds the most recent buffered range that comes before or includes the current time and uses its end point to calculate the total buffered percentage. This ensures we capture the relevant buffered duration relative to the current playback position, even in scenarios where buffering is discontinuous or occurs in segments.
+    for (let i = 0; i < buffered.length; i++) {
```

Comment:
This is still not clear to me but there are better ways to iterate an array in reverse. For example, 
```
    for (let i = buffered.length - 1; i >= 0; --i) {
```